jax/_src/numpy/polynomial.py

```diff
@@ -272,15 +272,14 @@ def polyfit(x: ArrayLike, y: ArrayLike, deg: int, rcond: float | None = None,
     Vbase = linalg.inv(dot(lhs.T, lhs))
     Vbase /= outer(scale, scale)
     if cov == "unscaled":
-      fac = 1
+      fac = array([1.0])
     else:
       if len(x_arr) <= order:
         raise ValueError("the number of data points must exceed order "
                             "to scale the covariance matrix")
       fac = resids / (len(x_arr) - order)
-      fac = fac[0] #making np.array() of shape (1,) to int
     if y_arr.ndim == 1:
-      return c, Vbase * fac
+      return c, Vbase * fac[0]
```

Comment:
This change doesn't seem to solve the issue, and instead seems to introduce a new broadcasting error. It might be worth comparing to the implementation in numpy, while remembering that JAX is more strict about rank promotion ([see docs here](https://jax.readthedocs.io/en/latest/rank_promotion_warning.html)).

Reply:
Thanks for pointing to the `numpy_rank_promotion` docs. It looks like the extra `num_rhs` tests I added are failing for this PR due to the broadcast error, but that is also happening on the main branch for any `polyfit` with multiple right hand sides:


```python
import numpy as np; import jax; import jax.numpy as jnp

x = np.arange(10).astype('float32')
y = np.random.randn(x.shape[0], 4)

with jax.numpy_rank_promotion("error"):
    jnp.polyfit(x, y, 1)
```

This gives me `ValueError: Operands could not be broadcast together for true_divide on shapes (4, 2) (2,)` at the line `c = (c.T/scale).T  # broadcast scale coefficient`.

If I run
```python
print(np.polyfit(x, y, 1, cov=True))
print(jnp.polyfit(x, y, 1, cov=True))
```
this PR now gives the same result as Numpy, while the main branch only gives 1 covariance output.

Should I attempt to fix the broadcast problem? Or should I add the same decorator as the next test, `@jax.numpy_rank_promotion('allow')  # This test explicitly exercises implicit rank promotion.`, which causes it the test to pass?

Reply:
> Should I attempt to fix the broadcast problem? Or should I add the same decorator as the next test, `@jax.numpy_rank_promotion('allow')  # This test explicitly exercises implicit rank promotion.`, which causes it the test to pass?

I'm so sorry for missing this comment!!

I think it would be best to try to work out the broadcasting properly, otherwise users who don't have `numpy_rank_promotion` set to `'allow'` will hit this same problem. Another option might be to locally set `numpy_rank_promotion` inside the implementation, but I think it's probably simple enough to get the broadcasting right explicitly. What do you think about that?

Sorry again for the slow response!!